include/aca_grpc.h

```diff
@@ -21,55 +21,88 @@
 
 using namespace alcor::schema;
 using grpc::Server;
+using grpc::ServerAsyncReader;
+using grpc::ServerAsyncReaderWriter;
+using grpc::ServerAsyncResponseWriter;
 using grpc::ServerBuilder;
+using grpc::ServerCompletionQueue;
 using grpc::ServerContext;
 using grpc::ServerReader;
 using grpc::ServerReaderWriter;
-using grpc::ServerAsyncResponseWriter;
-using grpc::ServerAsyncReader;
-using grpc::ServerAsyncReaderWriter;
-using grpc::ServerCompletionQueue;
 using grpc::ServerWriter;
 using grpc::Status;
 
 class GoalStateProvisionerAsyncServer {
   public:
+  /*
+    Base class that represents a gRPC call.
+    When you have a new kind of rpc, add the corresponding enum to CallType
+  */
+  struct AsyncGoalStateProvionerCallBase {
+    enum CallType { PUSH_NETWORK_RESOURCE_STATES, PUSH_GOAL_STATE_STREAM };
+    enum CallStatus { INIT, SENT };
+    CallStatus status_;
+    CallType type_;
+    grpc::ServerContext ctx_;
+  };
+
+  //  struct for PushNetworkResourceStates, which is a unary gRPC call
+  //  when adding a new unary rpc call, create a new struct just like PushNetworkResourceStatesAsyncCall
+  struct PushNetworkResourceStatesAsyncCall : public AsyncGoalStateProvionerCallBase {
```

Comment:
Why struct vs. class?

Reply:
I was following an online example, in which it uses struct. I guess we can use either one of them.